components/bellsound/ConvertStep.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+
+import { useRef, useState } from "react"
+import { CommonProps } from "./BellSoundWalkthrough"
+import { FFmpeg } from "@ffmpeg/ffmpeg"
+import { fetchFile, toBlobURL } from "@ffmpeg/util"
+import WalkthroughButton from "./WalkthroughButton"
+
+export default function ConvertStep({ onDismiss, selectedFile, onConversionCompleted }: CommonProps & {
+    selectedFile: File,
+    onConversionCompleted: (convertedFile: Uint8Array) => void,
+}) {
+    const [ffmpegLog, setFfmpegLog] = useState<string>("")
+    const [showLog, setShowLog] = useState<boolean>(false)
+    const [converting, setConverting] = useState<boolean>(false)
+    const ffmpegRef = useRef<FFmpeg>(new FFmpeg())
+
+    const log = (message: string) => {
+        console.log(`[ConvertStep] ${message}`)
+        setFfmpegLog((prev) => prev + message + "\n")
+    }
+
+    const loadFfmpeg = async () => {
+        const baseURL = "https://unpkg.com/@ffmpeg/core@0.12.2/dist/umd"
+        const ffmpeg = ffmpegRef.current
+        if (ffmpeg.loaded) return
+
+        ffmpeg.on("log", ({ message }) => {
+            log(message)
+        })
+
+        log("Loading ffmpeg...")
+
+        await ffmpeg.load({
+            coreURL: await toBlobURL(
+                `${baseURL}/ffmpeg-core.js`,
+                "text/javascript",
+            ),
+            wasmURL: await toBlobURL(
+                `${baseURL}/ffmpeg-core.wasm`,
+                "application/wasm",
+            ),
+        })
+    }
+
+    const fromHexString = (hexString: string) =>
+        Uint8Array.from(hexString.match(/.{1,2}/g)!.map((byte: string) => parseInt(byte, 16)))
+
+    const applyHeader = (file: Uint8Array): Uint8Array => {
+        const headerStart = fromHexString("564D5F534F554E44FFFFFFFF015858585800585858585858")
+        const fileWithHeader = new Uint8Array(headerStart.length + file.length + 4)
+
+        fileWithHeader.set(headerStart, 0)
+
+        const fileSize = file.byteLength
+        fileWithHeader.set([fileSize & 0xFF, (fileSize >> 8) & 0xFF, (fileSize >> 16) & 0xFF, (fileSize >> 24) & 0xFF], headerStart.length)
+
+        fileWithHeader.set(file, headerStart.length + 4)
+        return fileWithHeader
+    }
+
+    const startConversion = async () => {
+        if (converting) return
+        setConverting(true)
+
+        await loadFfmpeg()
+        const ffmpeg = ffmpegRef.current
+
+        log("Loading file into memory...")
+        await ffmpeg.writeFile(selectedFile.name, await fetchFile(selectedFile))
+
```

Comment:
```suggestion

        // Calculate the sample rate based on the length of the song
        const guessedSampleRate = Math.round(400_000 / selectedFile.size / 2.2 * 4000)
        log(`Guessed sample rate: ${guessedSampleRate}`)

```

Reply:
Don't ask me how I came up with this hackory sorcery shit I just put some things together and seemed to work... xD

Reply:
Seems to guess it right for everything tried a 50 MB file guessed 14 its low but its the whole Shrek movie and it fits!!!!!!

Reply:
Imma try increasing it a little see what the limit is to finetune the special algorithm I designed xD

Reply:
Okay 6000 seems to be about the sweat spot so most different kinds all work

Reply:
We should look into how it works exactly instead of this magic stuff but works fine for now